src/package_tracking/include/package_tracking/package_edge_detection.h

```diff
@@ -0,0 +1,60 @@
+//
+// Created by Shahwaz Khan on 2/19/21.
+//
+
+//// This node detects the edge of the object which is at the closest distance from the origin of the realsense camera.
+//// The z-axis emerges from the lens of the camera, the y-axis always points downwards and the x-axis towards right.
+//// RIGHT HAND THUMB RULE
+
+#ifndef SRC_PACKAGE_EDGE_DETECTION_H
+#define SRC_PACKAGE_EDGE_DETECTION_H
+
+#include "point_cloud_utilities/pcl_utilities.h"
+
+
+typedef pcl::PointCloud<pcl::PointXYZRGB> PointCloudT;
+
+class PackageTracking{
+
+protected:
+    /* data */
+    std::string file_name_;
+    std::string file_path_ = "/home/shahwaz/catkin_ws/src/my_pcl_tutorials/write_data";
+
+public:
+    PackageTracking();
+    ros::NodeHandle nh_;
+
+//  Point cloud tracking members
+    ros::Publisher package_cloud_pub_;
+    std::vector<double> cam_box_limits_;
+    PointCloudT::Ptr box_filtered_cloud_ptr_;
+    PointCloudT::Ptr mesh_cloud_ptr_;
+    PointCloudT::Ptr final_cloud_created;
+
+    double tolerance{};
+
+    PointCloudT cloudIn_xz;
+
+
+
+
+
+//  Camera members
```

Comment:
Too many empty lines.

Reply:
Yes brother, I have to format the code. There's a keyboard shortcut for that in CLiOn. I'll do that and push the code later.